DirectX11/HackerContext.cpp

```diff
@@ -2647,31 +2647,34 @@ STDMETHODIMP_(void) HackerContext::IASetIndexBuffer(THIS_
 	/* [annotation] */
 	__in  UINT Offset)
 {
-	FrameAnalysisLog("IASetIndexBuffer(pIndexBuffer:0x%p, Format:%u, Offset:%u)\n",
-			pIndexBuffer, Format, Offset);
-	FrameAnalysisLogResourceHash(pIndexBuffer);
-
 	mOrigContext->IASetIndexBuffer(pIndexBuffer, Format, Offset);
 
-	// When hunting, save this as a visited index buffer to cycle through.
-	if (pIndexBuffer && !G->mDataBuffers.empty() && G->hunting == HUNTING_MODE_ENABLED) {
+	// Save the current index buffer hash for indexbufferfilter in ShaderOverride.
+	if (pIndexBuffer && !G->mDataBuffers.empty()) {
```

Comment:
I think I have a better idea. Instead of looking for the hash in our own map, it is better off just store the hash into the ID3D11Buffer object itself using SetPrivateData. This way it will be super efficient, but need some more changes. In fact by using this approach we don't even need mDataBuffers at all. I will try if it actually work.